### PR TITLE
Skip validation for early epochs

### PR DESCRIPTION
**Pull request by bstaber**
## What does this PR do?

Adds a performance trick to the documentation. It proposes a solution to skip validation for early epochs. It can especially useful when validation is performed after each training batch but we are not interested in early validation.

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [X] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [X] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [X] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [X] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [X] Is this pull request ready for review? (if not, please submit in draft mode)
 - [X] Check that all items from **Before submitting** are resolved
 - [X] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/benchmarking/performance.rst [hunk 1]
```diff
@@ -181,3 +181,21 @@ read `this <https://pytorch.org/docs/master/optim.html#torch.optim.Optimizer.zer
...
+Note: doesn’t work nicely with ``EarlyStopping``.
+
+.. code-block:: python
+
+    def on_train_epoch_start(self):
+        if self.trainer.current_epoch >= self.num_burnin_epochs:
+            self.trainer.limit_val_batches = 1.0
+        else:
+            self.trainer.limit_val_batches = 0
+
```
Comment (ananthsub) [inline docs/source/benchmarking/performance.rst:201]: this can be quite risky updating the trainer properties like this, especially if the trainer derives state from these arguments before the module changes them here. it can quickly result in things falling out of sync. 

@awaelchli @carmocca @justusschock : this is supported on the training batch loop, but not the evaluation loop: https://github.com/PyTorchLightning/pytorch-lightning/blob/d4d5418cc4086a7cb8ecc18a42e0eb109b3b7859/pytorch_lightning/loops/training_batch_loop.py#L81-L91

any reason why? if the lightning module had a way to signal to the trainer to short-circuit the val/test epoch, resetting the trainer properties wouldn't be needed.

Comment (awaelchli) [reply docs/source/benchmarking/performance.rst:201]: for this usecase it feels more natural to introduce an enable/disable mechanism on the loops, for example:

```
# could be method or property setter
self.trainer.fit_loop.validation_loop.disable = self.trainer.current_epoch < self.num_burnin_epochs
```

because if the intention is to skip the val loop entirely, then returning -1 from on_val_batch_start is not fully satisfactory as the on_validation_epoch_start / end etc. hooks would still run.

Comment (bstaber) [reply docs/source/benchmarking/performance.rst:201]: Thanks for your inputs. Do you think it's still worth mentioning this trick in the doc ?

Comment (carmocca) [reply docs/source/benchmarking/performance.rst:201]: There's no need for a disable mechanism if loops are customizable, right?

Comment (awaelchli) [reply docs/source/benchmarking/performance.rst:201]: The question is whether we want to recommend the user to implement/subclass a whole new loop class just to add a disable condition to it, vs. exposing a method.
